SwiftWisdomTests/StandardLibrary/String/String+IndexingTests.swift

```diff
@@ -2,6 +2,30 @@ import XCTest
 
 class String_IndexingTests: XCTestCase {
     
+    func testRangeSubscriptReturnsProperString() {
+        XCTAssertEqual("hello"[0..<0], "")
+        XCTAssertEqual("hello"[0..<1], "h")
+        XCTAssertEqual("hello"[0..<5], "hello")
```

Comment:
Do we also need some kind of out-of-bounds test (i.e., `XCTAssertSOMETHING("hello"[0..<6], "hello")`) to ensure that the desired behavior (whatever that is - return the full string, throw a specific error, probably NOT seg-fault) doesn't break in the future?